OpenBCISample.js

```diff
@@ -0,0 +1,200 @@
+
+        var channelData = function () {
+            var out = {};
+            var count = 1;
+            for (var i = 2; i <= numberOfBytes - 10; i += 3) {
+                //console.log('\tDataBuf is is: ' + dataBuf.slice(i, i + 3).toString('hex'));
+                //var newNumber = self.interpret24bitAsInt32(dataBuf.slice(i, i + 3));
+                //out[count] = newNumber * SCALE_FACTOR_CHANNEL;
+                out[count] = scaleData ? self.interpret24bitAsInt32(dataBuf.slice(i, i + 3)) * SCALE_FACTOR_CHANNEL : self.interpret24bitAsInt32(dataBuf.slice(i, i + 3));
+                //console.log("in" + dataBuf.slice(i,i+3));
+                //console.log(out[count]);
+                count++;
+            }
+            return out;
+        };
+
+        var auxData = function () {
+            var out = {};
+            var count = 0;
+            for (var i = numberOfBytes - 7; i < numberOfBytes - 1; i += 2) {
+                out[count] = scaleData ? self.interpret16bitAsInt32(dataBuf.slice(i, i + 2)) * SCALE_FACTOR_ACCEL : self.interpret16bitAsInt32(dataBuf.slice(i, i + 2));
+                count++;
+            }
+            return out;
+        };
+
+        return {
+            startByte: dataBuf[0], //byte
+            sampleNumber: dataBuf[1], //byte
+            channelData: channelData(), // multiple of 3 bytes
+            auxData: auxData(), // 6 bytes
+            stopByte: dataBuf[numberOfBytes - 1] //byte
+        }
+    },
+    debugPrettyPrint: function(sample) {
+        console.log('-- Sample --');
+        console.log('---- Start Byte: ' + sample.startByte);
+        console.log('---- Sample Number: ' + sample.sampleNumber);
+        for(var i = 0; i < 8; i++) {
+            console.log('---- Channel Data ' + i + ': ' + sample.channelData[i]);
+        }
+        for(var j = 0; j < 3; j++) {
+            console.log('---- Aux Data ' + j + ': ' + sample.auxData[j]);
+        }
+        console.log('---- Stop Byte: ' + sample.stopByte);
+
+
+    },
+    scaleFactorAux: SCALE_FACTOR_ACCEL,
+    scaleFactorChannel: SCALE_FACTOR_CHANNEL,
+    sampleMaker: function(length) {
+        var data = new Buffer(0);
+        return function (buffer) {
+            data = Buffer.concat([data, buffer]);
+            while (data.length >= length) {
+                var out = data.slice(0, length);
+                data = data.slice(length);
+            }
+        };
+    },
+    interpret24bitAsInt32: function(threeByteBuffer) {
+        //const maskForNegativeNum = (255 << (3 * 8));
+        //const maskForPositiveNum = 255 | (255 << 8) | (255 << 16);
+
+        var prefix = 0;
+
+        if(threeByteBuffer[0] > 127) {
+            //console.log('\t\tNegative number');
+            prefix = 255;
+        }
+
+        var newInt = (prefix << 24 ) | (threeByteBuffer[0] << 16) | (threeByteBuffer[1] << 8) | threeByteBuffer[2];
+
+        return newInt;
+        //// 3byte int in 2s compliment
```

Comment:
yeah, sure. just try to leave some comments along the way for clarity. some of your commented out comments were helpful with understanding the code